src/main/scala/edu/cmu/cs/obsidian/codegen/CodeGenYul.scala

```diff
@@ -23,16 +23,35 @@ object CodeGenYul extends CodeGenerator {
     var stateEnumMapping: Map[String, Int] = Map() // map from state name to an enum value
     var stateEnumCounter: Int = 0 // counter indicating the next value to assign since we don't know the total num of states
 
+    /**
+      * as we translate transactions, we note their name and the number of things they return. then
+      * when we translate a call to them we can produce the right number of temporary variables to
+      * capture the result.
+      *
+      * todo: this may be too naïve but it will work for now
+      */
+    var fnNamesToReturnArity: Map[String, Int] = Map()
```

Comment:
I think we need to remove this and replace it with lookups in the context. Otherwise we'll have an ordering dependency.

Reply:
got it; i fixed this in https://github.com/mcoblenz/Obsidian/pull/331/commits/e0570da498bbec0b2c6c93ca4849e38f15321ed4 and https://github.com/mcoblenz/Obsidian/pull/331/commits/84e302ff13d644d92502141dd4df0af015e7a9f1. in retrospect this is all a bit overwrought, but it works.